components/camel-cdi/pom.xml

```diff
@@ -37,10 +37,37 @@
         <label>java</label>
         <title>CDI</title>
 
+        <camel.osgi.provide.capability>
+            osgi.cdi.extension;osgi.cdi.extension="org.apache.camel.cdi";
+                version:Version="${camel.osgi.version}";
+                aries.cdi.extension.bean.classes:List&lt;String&gt;="org.apache.camel.cdi.CdiCamelFactory,org.apache.camel.cdi.CdiEventComponent";
+                    uses:="javax.annotation,javax.enterprise.event,javax.enterprise.inject.spi",
+            osgi.service;objectClass:List&lt;String&gt;="javax.enterprise.inject.spi.Extension",
+                osgi.service;objectClass:List&lt;String&gt;="javax.enterprise.inject.spi.Extension";
+                    uses:="javax.enterprise.inject.spi";
+                    osgi.cdi.extension="org.apache.camel.cdi";
+                service.scope=prototype;service.vendor="Apache Software Foundation";
+                version:Version="${camel.osgi.version}";
+                effective:=active,
+            osgi.serviceloader;osgi.serviceloader="javax.enterprise.inject.spi.Extension";
+                register:="org.apache.camel.cdi.CdiCamelExtension";
+                    uses:="javax.enterprise.inject.spi";
+                    osgi.cdi.extension="org.apache.camel.cdi";
+                    service.scope=prototype;
+                    service.vendor="Apache Software Foundation";
+                version:Version="${camel.osgi.version}"
+        </camel.osgi.provide.capability>
+        <camel.osgi.require.capability>
+            osgi.extender;filter:="(&amp;(osgi.extender=osgi.cdi)(version>=1.0.0)(!(version>=2.0.0)))",
```

Comment:
Is this correct, it seems to be cdi 1.x only? We are CDI 2.0 onwards only for Camel 3.x.

Reply:
Yes, osgi.cdi v1 (= cdi 2)

Reply:
OSGi is lovely ;)

All that osgi metadata above is written by hand and can get out of cruft as its woodo for most people what has to be there.

Reply:
You can make it generated with bnd 5 using osgi.annotations (which is fine for camel since scope would be provided) but not sure it is better ;).
At the end I don't care much how but I'd like to be able to use camel-cdi with osgi-cdi :).

Reply:
Yeah sure I would assume you have tried it so its safe to merge

Reply:
Tried in a fake osgi container (winegrower) but happy to get some real osgi container test.
That said it cant be worse than today evennif broken ;)